alsa/tinyalsa_hal.c

```diff
@@ -291,8 +291,11 @@ static int set_route_by_array(struct mixer *mixer, struct route_setting *route,
     i = 0;
     while (route[i].ctl_name) {
         ctl = mixer_get_ctl_by_name(mixer, route[i].ctl_name);
-        if (!ctl)
-            return -EINVAL;
+        if (!ctl) {
+          ALOGW("Can't set mixer ctl value for missing control %s", route[i].ctl_name);
+          i++;
+          continue;
+        }
 
```

Comment:
Hi Nabil,

Thank you once again for sharing this patch.
Before merging it, can you update that patch to have the proper indentation (1 tab = 4 spaces).

Thanks!